File: TOMIMSV4/Entity/Interface/Pedido_Traslado/Pedido_Traslada_Det_Lotes/clsBeI_nav_ped_traslado_det_lote.vb
Public Class clsBeI_nav_ped_traslado_det_lote
    Implements ICloneable

    Public Property NoEnc() As String = ""
    Public Property Line_No() As Integer = 0
    Public Property No() As String = ""
    Public Property Batch_No() As String = ""
    Public Property Serial_No() As String = ""
    Public Property Expiration_Date() As Date = Date.Now
    Public Property Quantity_Base() As Double = 0.0
    Public Property Variant_Code() As String = ""
    Public Property WhsFrom() As String = ""
    Public Property WhsTo() As String = ""
    Public Property Fec_Agr() As Date = Date.Now

    Sub New()
        '#EJC: Add your constructor here... 
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Pedido/clsBeI_nav_ped_traslado_det_lote.cs
namespace WMS.EntityCore.Pedido
{
    public class clsBeI_nav_ped_traslado_det_lote : ICloneable
    {
        public string NoEnc { get; set; } = string.Empty;
        public int Line_No { get; set; } = 0;
        public string No { get; set; } = string.Empty;
        public string Batch_No { get; set; } = string.Empty;
        public string Serial_No { get; set; } = string.Empty;
        public DateTime Expiration_Date { get; set; } = DateTime.Now;
        public double Quantity_Base { get; set; } = 0.0;
        public string Variant_Code { get; set; } = string.Empty;
        public string WhsFrom { get; set; } = string.Empty;
        public string WhsTo { get; set; } = string.Empty;
        public DateTime Fec_Agr { get; set; } = DateTime.Now;

        public clsBeI_nav_ped_traslado_det_lote()
        {
            // Constructor vacío
        }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}